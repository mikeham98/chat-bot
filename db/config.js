export const currentUser = {
    userId: 1,
    userName: "mikeham98"
};

// the conversation id is the key for each object
export const conversations = {
    "1": {
        userId: "bot1",
        userName: "travel_bot",
        messages: [{
            body: 'Where would you like to go to?',
        }, {
            body: 'What a lovely destination, how long would you like to go for?'
        }, {
            body: 'Brilliant, how many people shall I book for?'
        }, {
            body: 'Perfect, when would you like to go?'
        }, {
            body: 'What airport would you like to fly from?'
        }, {
            body: 'We have two options available for London'
        }, {
            body: 'What time would you like to fly out'
        }, {
            body: 'When would you like your return flight to be at'
        }, {
            body: 'So just to confirm, you want to go to Florida for 2 weeks from August 12th (fly out 10am) August 26 (return flight 2pm) with 3 people'
        }, {
            body: 'Brilliant, that is all booked for you, have an amazing holiday!'
        }]
    },
    "2": {
        userId: "bot2",
        userName: "Pizza_bot",
        messages: [{
            body: 'Sure thing! what size pizza would you like?',
        }, {
            body: 'Would you like any toppings?'
        }, {
            body: 'We will get working on it right away',
            media: [{
                type: 'gif',
                source: 'https://media.giphy.com/media/1108D2tVaUN3eo/giphy.gif'
            }]
        }, {
            body: 'We will get working on it right away',
            media: [{
                type: 'gif',
                source: 'https://media.giphy.com/media/1108D2tVaUN3eo/giphy.gif'
            }]
        }, {
            delay: 10 * 1000,
            body: 'Voila, your pizza has been made and is out for delivery',
            media: [{
                type: 'gif',
                source: 'https://media.giphy.com/media/QtoVsU8wdqbAs/giphy.gif'
            }]
        }]
    },
    "3": {
        userId: "bot3",
        userName: "web_search_bot",
        messages: [{
            body: 'Sure thing, what can I help you with',
        }, {
            body: 'I have found a good article for you, hope this helps',
            media: [{
                type: 'link',
                source: 'https://waitbutwhy.com/2015/08/how-and-why-spacex-will-colonize-mars.html'
            }]
        }, {
            body: 'You`re welcome, is there anything else I can help you with?'
        }, {
            body: 'Okay, feel free to ask for any help in the future, have a good day'
        }]
    }
};