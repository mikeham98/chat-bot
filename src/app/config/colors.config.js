const flatRed = '#f4244e';
const flatOrange = '#f87b2b';
const flatYellow = '#F1C40E';
const flatGreen = '#30CB71';
const flatTeal = '#1ccb9e';
const flatBlue = '#3598DA';
const flatLavender = '#6872FF';
const flatPurple = '#9A58B9';

export const flatColours = [
    flatRed,
    flatOrange,
    flatYellow,
    flatGreen,
    flatTeal,
    flatBlue,
    flatLavender,
    flatPurple
];