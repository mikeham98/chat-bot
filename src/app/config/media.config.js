export const gif = 'gif';
export const img = 'img';
export const link = 'link';