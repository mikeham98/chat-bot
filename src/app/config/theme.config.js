export const lightTheme = 'light';
export const darkTheme = 'dark';

export const oppositeTheme = (theme) => {
    if(theme === lightTheme) {
        return darkTheme;
    }
    return lightTheme;
};