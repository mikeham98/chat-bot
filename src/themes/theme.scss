@import 'colors';

.darkTheme {
  color: white;
  background-color: $flat-black;
  .messageOptionWrapper .messageOption, .leftMessage .message {
    border-color: $jet-black;
  }
  .messageOptionWrapper .messageOptionDisabled, .messageOptionWrapper .messageOptionDisabled:hover {
    color: $dark-gray;
  }

  .mediaLinkPreviewWrapper .mediaLinkPreviewMeta .mediaLinkPreviewDescription, .leftMessage .message {
    color: white;
  }
  .conversationWrapper,
  .sideBarMenu,
  .mediaLinkPreviewWrapper,
  .mediaImagePreviewImage,
  .headerLayout {
    border-color: $dark-gray;
  }
  .mediaLinkPreviewWrapper .mediaLinkPreviewMeta, .leftMessage .message {
    background-color: $jet-black;
  }
  .messageFormWrapper, .headerLayout, .sideBarMenu, .messageOptionWrapper, .messageContainer {
    background-color: $flat-black;
  }
  .messageFormWrapper {
    input {
      color: white
    }
  }
}