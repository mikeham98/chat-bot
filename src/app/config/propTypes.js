import PropTypes from 'prop-types';
import {gif, img, link} from "./media.config";

export const OptionPropTypes = PropTypes.shape({
    id: PropTypes.number.isRequired,
    option: PropTypes.string.isRequired,
});

export const MediaObjectPropTypes = PropTypes.shape({
    id: PropTypes.number.isRequired,
    src: PropTypes.string.isRequired,
    type: PropTypes.oneOf([gif, img, link]).isRequired,
    imagePreview: PropTypes.string,
    title: PropTypes.string,
    description: PropTypes.string,
});

export const MediaPropTypes = PropTypes.arrayOf(MediaPropTypes);

export const OptionsPropTypes = PropTypes.arrayOf(OptionPropTypes);

export const MessagesPropTypes = PropTypes.arrayOf(PropTypes.shape({
    content: PropTypes.shape({
        body: PropTypes.string.isRequired,
        media: MediaPropTypes,
        options: OptionsPropTypes,
    }),
    recipient: PropTypes.shape({
        userId: PropTypes.oneOfType([
            PropTypes.string,
            PropTypes.number,
        ]),
        userName: PropTypes.string
    }),
    createdAt: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.instanceOf(Date)
    ]).isRequired,
    createdBy: PropTypes.shape({
        userId: PropTypes.oneOfType([
            PropTypes.string,
            PropTypes.number,
        ]),
        userName: PropTypes.string,
    }).isRequired
}));