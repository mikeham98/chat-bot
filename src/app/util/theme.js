import {darkTheme, lightTheme} from '../config/theme.config';

export const oppositeTheme = (theme) => {
    if(theme === lightTheme) {
        return darkTheme;
    }
    return lightTheme;
};