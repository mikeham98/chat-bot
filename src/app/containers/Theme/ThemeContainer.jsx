import React from 'react';
import styles from '../../../themes/theme.scss';
import {connect} from 'react-redux';
import {darkTheme} from "../../config/theme.config";

const ThemeContainer = ({theme, children}) => {
    console.log(theme)
    console.log(darkTheme)
    return (
        <div className={theme === darkTheme ? styles.darkTheme : ''}>
            {children}
        </div>
    )
};

const mapStateToProps = (state) => {
    return {
        theme: state.settings.theme
    }
};

export default connect(mapStateToProps)(React.memo(ThemeContainer))