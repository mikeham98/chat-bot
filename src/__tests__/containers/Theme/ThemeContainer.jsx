import React from 'react';
import {shallow} from 'enzyme';
import {ThemeContainer} from '../../../app/containers/Theme/ThemeContainer';
import {lightTheme, darkTheme} from '../../../app/config/theme.config';

describe('ThemeContainer', () => {
    let shallowedWrapper;
    let props = {};
    const wrapper = () => {
        if (!shallowedWrapper) {
            shallowedWrapper = shallow(<ThemeContainer {...props}/>);
        }
        return shallowedWrapper;
    };

    beforeEach(() => {
        props = {
            theme: lightTheme,
            children: <div>Children</div>
        };
        shallowedWrapper = undefined;
    });

    describe('render', () => {
        it('should return a light theme (empty class name) with a div with text of Children', () => {
            expect(wrapper()).toMatchSnapshot();
        });
        it('should return a dark theme with a div with text of Children', () => {
            props.theme = darkTheme;
            expect(wrapper()).toMatchSnapshot();
        });
    });
});