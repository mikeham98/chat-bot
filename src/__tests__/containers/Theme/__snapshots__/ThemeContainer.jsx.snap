// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeContainer render should return a dark theme with a div with text of Children 1`] = `
<div
  className="darkTheme"
>
  <div>
    Children
  </div>
</div>
`;

exports[`ThemeContainer render should return a light theme (empty class name) with a div with text of Children 1`] = `
<div
  className=""
>
  <div>
    Children
  </div>
</div>
`;

exports[`ThemeContainer render should return a light theme with a div with text of Children 1`] = `
<div
  className=""
>
  <div>
    Children
  </div>
</div>
`;
